pkg/registry/core/shoot/strategy_test.go

```diff
@@ -25,6 +28,73 @@ import (
 	"k8s.io/apimachinery/pkg/labels"
 )
 
+var _ = Describe("Strategy", func() {
+	Context("PrepareForUpdate", func() {
+		Context("pod pids limit", func() {
+			It("should enforce the minimum limit value", func() {
+				var tooSmallValue int64 = 10
+
+				shoot := newShoot("foo")
+				shoot.Spec.Kubernetes.Kubelet = &core.KubeletConfig{
+					PodPIDsLimit: &tooSmallValue,
+				}
+				shoot.Spec.Provider.Workers = []core.Worker{
+					{
+						Kubernetes: &core.WorkerKubernetes{
+							Kubelet: &core.KubeletConfig{
+								PodPIDsLimit: &tooSmallValue,
+							},
+						},
+					},
+				}
+
+				oldShoot := newShoot("foo")
+				oldShoot.Spec.Kubernetes.Kubelet = shoot.Spec.Kubernetes.Kubelet.DeepCopy()
+				oldShoot.Spec.Provider.Workers = []core.Worker{shoot.Spec.Provider.Workers[0]}
```

Comment:
```suggestion
				oldShoot.Spec.Provider.Workers = []core.Worker{shoot.Spec.Provider.Workers[0].DeepCopy()}
```
Can we use `DeepCopy()` to make sure the tested code is not operating on the exact same object?

Reply:
`Workers[0]` is of type `core.Worker`, hence, it doesn't have a `DeepCopy` function but is already a copy.

Reply:
Sorry, I don't quite get it.
`core.Worker` seems to have a `DeepCopy` function: https://github.com/gardener/gardener/blob/13d359fa934466fd8ed2452e8042ebd646d9260c/pkg/apis/core/zz_generated.deepcopy.go#L3257-L3265

Additionally when I debug the test, it shows me the same pointers for `shoot.Spec.Provider.Workers[0].Kubernetes` and `oldShoot.Spec.Provider.Workers[0].Kubernetes`, so I don't think the nested objects are copied.  
For the `core.Worker` objects, you are right, because they are just `struct` values and no pointers. But the copy of the `struct` value still contains the same pointer values for its fields like for example `.Kubernetes`.

Reply:
You're right, it just copies the first level but not transitively. OK, fixed.